### ISSUE TITLE
Add input for stestr run --concurrency flag for fractional nprocs

### ISSUE THREAD

**Issue by mtreinish** [mtreinish/stestr#202]
Right now the --concurrency flag for stestr run takes in an integer, if that value is 0 it runs nproc workers, if that value is greater than 0 it runs that many workers. But we don't have a way to specify a value that is a fraction of nproc easily (like nproc/2 or nproc/4). This can be useful for some test suites, especially if they're very resource intensive, where running nproc workers isn't viable but you still want the number of workers to scale dynamically based on the size of the machine.

There are 2 ways I can see handling this first we also take a string in for concurrency and define an input for the fractions. Like `stestr run --concurrency 'n/2'` will run with nprocs / 2 workers. Or we could use the negative space and stick with integers to define the same thing, so something like `stest run --concurrency -2` will run with nprocs / 2 workers.

**Comment by mtreinish** [comment]
@masayukig do you have any thoughts on this?

**Comment by masayukig** [comment]
yeah, I think it's a good feature. I actually don't need like that option so far personally. But it makes sense to me.

So, regarding the option argument, I'm fine with either way. But I prefer `-2` style because it's simpler.

I was also think like `--concurrency '0.5'` but I feel that would cause a confusion and inconsistency.

**Comment by mtreinish** [comment]
Well it mostly came up for me because of: https://github.com/Qiskit/qiskit-terra/pull/1228 where we're hitting occasionally memory issues. I'd rather set it to nproc/2 or nproc/4 rather than hard coding it to 2 workers.

**Comment by mtreinish** [comment]
As for the specific input definition I'm fine with anything as long as we make it clear. I only suggested '-2' or 'n/2' because I thought the implementation would be simpler. If we take a float my concern is that someone will do something like `--concurrency 0.124332678212341`. Granted we will always have the cast the result as an `int` so maybe that isn't a problem.

**Comment by masayukig** [comment]
Hrm. I actually don't have a strong opinion here. If we know the nproc already, the subtraction might work well. But I also think, if we know that, can we specify the concurrency directly? So, I guess we use this feature when the nproc is ambiguous. If so, I feel treating the concurrency as a fraction might be better.

And, I'm curious about a corner case of the subtraction. What will happen when nproc =< abs(concurrency) && 0 > concurrency? For example, nproc=32 and --concurrency=-32.
Of course, I think we can treat this just no concurrency in this case, though.

**Comment by mtreinish** [comment]
Yeah, I think treating it as a fraction is probably better. I just wrote that down because it was an idea I had related to this, after thinking more I agree with subtraction is not a good idea.

As for that corner case yeah there are probably 2 ways to handle it, either we exit with an error code and say it's not a valid value, or we treat it as running serially. Actually, this made me think what happens today if we pass a negative number to `--concurrency`, so I ran a test:

```
stestr run --concurrency -23
list index out of range
```

We probably should change that error message to be more descriptive.

**Comment by masayukig** [comment]
heh, yeah. Oh, did you already start to implement this?
